src/m365/pp/commands/card/card-get.spec.ts

```diff
@@ -0,0 +1,196 @@
+    "statuscode": 1,
+    "remixsourceid": null,
+    "sizes": null,
+    "_owningteam_value": null,
+    "coowners": null,
+    "_createdby_value": "7d48edd3-69fd-ec11-82e5-000d3ab87733",
+    "_ownerid_value": "7d48edd3-69fd-ec11-82e5-000d3ab87733",
+    "publishsourceid": null,
+    "timezoneruleversionnumber": null,
+    "iscustomizable": {
+      "Value": true,
+      "CanBeChanged": true,
+      "ManagedPropertyLogicalName": "iscustomizableanddeletable"
+    },
+    "owninguser": {
+      "azureactivedirectoryobjectid": "88e85b64-e687-4e0b-bbf4-f42f5f8e574c",
+      "fullname": "Contoso Admin",
+      "systemuserid": "7d48edd3-69fd-ec11-82e5-000d3ab87733",
+      "ownerid": "7d48edd3-69fd-ec11-82e5-000d3ab87733"
+    }
+  };
+  //#endregion
+
+  let log: string[];
+  let logger: Logger;
+  let loggerLogSpy: sinon.SinonSpy;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    auth.service.connected = true;
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      appInsights.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.CARD_GET), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('defines correct properties for the default output', () => {
+    assert.deepStrictEqual(command.defaultProperties(), ['name', 'cardid', 'publishdate', 'createdon', 'modifiedon']);
+  });
+
+  it('retrieves specific card', async () => {
+    sinon.stub(request, 'get').callsFake(async opts => {
+      if ((opts.url === `https://api.bap.microsoft.com/providers/Microsoft.BusinessAppPlatform/environments/4be50206-9576-4237-8b17-38d8aadfaa36?api-version=2020-10-01&$select=properties.linkedEnvironmentMetadata.instanceApiUrl`)) {
+        if (opts.headers &&
+          opts.headers.accept &&
+          (opts.headers.accept as string).indexOf('application/json') === 0) {
+          return envResponse;
+        }
+      }
```

Comment:
We should rather mock the `powerPlatform.getDynamicsInstanceApiUrl` function than mocking the request it is making.
We expect a result of the util function and don't really know/care how it is retrieved.

Same for other tests